test/integration2/dialects/oracledb.spec.js

```diff
@@ -1,5 +1,6 @@
 const { expect } = require('chai');
 const { getAllDbs, getKnexForDb } = require('../util/knex-instance-provider');
+const oracledb = require('oracledb');
```

Comment:
why need this ? you have the db below, in "dbs" variable

Reply:
@OlivierCavadenti correct me if I'm wrong, but from what I can tell, `getAllDbs`, which is what populates the `dbs` variable, just returns an array of strings indicating the names of all databases that have been specified to test (since it reads process.env.DB or a default array of all database client names). I'm not sure how that would give me what I am needing in this test. What I'm needing here is the oracle database driver package and not a string for the driver's name since I need to import some constants from the `node-oracledb` database driver package (exposed under the `oracledb` import name) itself since many of oracle's database driver options require use of constants defined in the package. (Line 62 shows where those constants are used.) As far as I can tell, `dbs` does not give me access to these constants.

Let me know if that clears up your question or if I'm perhaps misunderstanding something about the `dbs` variable.

Reply:
It seems it's that require that made the tests fails as I can see, can we avoid that ?

Reply:
Since Oracle driver by itself requires properly configured environment, every test that depends on it automatically becomes integration one. I would suggest extracting that particular test for Oracle into integration2

Reply:
@kibertoad I currently have this test in the `integration2` directory. Is there any additional configuration I need to do so that this is running as a proper integration test? @OlivierCavadenti I could rewrite this test to pass a different option, such as setting the `extendedMetaData` boolean option instead. It would not require importing any constants from the oracledb package, but it would still result in different behavior for the test query I am using and would still confirm that options set using `.options()` eventually make it to the oracle database driver for that specific query.

Reply:
@OlivierCavadenti @kibertoad I just realized that this test file was being included/executed even when other databases were specified to run, contrary to my expectation. Moving the test inside the `dbs.forEach` -> `describe` block prevented these tests from executing when oracle is not included in the databases to test for that run. I also moved the require call to the oracledb driver into the body of the test that needs it so that we don't run into that error about not having an oracle environment setup when that test is not being executed. I do need to `require` the oracle driver at some point since most of oracle's options require referencing some pre-defined values included in the driver for proper use. The driver name to require is also being resolved off of the knex instance as extra insurance so that the test won't include a driver package that is not installed.